src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.impl.MySqlConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest =
+                "INSERT INTO manufacturers (name, country)VALUES (?, ?);";
+        try (Connection connection = new MySqlConnectionUtil().getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't create a manufacturer: " + manufacturer.getName(), e);
```

Comment:
```suggestion
            throw new DataProcessingException(
                    "Can't create a manufacturer: " + manufacturer, e);
```

I think. it's better to log all info about manufacturer